### PR TITLE
Add starting real goto definition tests

### PR DESCRIPTION
**Pull request by mkaput**
These tests are very incomplete,
but they serve as a good basis for future work.

---

**Stack**:
- #123
- #122
- #121
- #119
- #118 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: tests/e2e/goto_definition.rs [hunk 1]
```diff
@@ -68,17 +74,26 @@ fn test_goto_members(
...
+                for location in locations {
+                    report.push_str("---\n");
+                    report.push_str(&peek_selection(&cairo, &location.range));
+                }
+            }
+            Some(GotoDefinitionResponse::Link(_)) => {
+                panic!("unexpected GotoDefinitionResponse::Link");
+            }
+            None => {
+                report.push_str("None");
```
Comment (piotmag769) [inline tests/e2e/goto_definition.rs:95]: Do we expect any failures here, shouldn't we panic here? Didn't find any "None" in tests

File: tests/test_data/goto/enum_variants.txt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <sel>Bar</sel>,
+
+//! > Goto definition #1
+        Foo::Ba<caret>r => {}
+---
+    <sel>Bar</sel>,
+
+//! > Goto definition #2
+Foo::Ba<caret>z => {}
+    <sel>Baz</sel>,
```
Comment (piotmag769) [inline tests/test_data/goto/enum_variants.txt:32]: Where does it come from?

Comment (piotmag769) [inline tests/test_data/goto/enum_variants.txt:32]: Where does it come from?

File: tests/test_data/goto/items.txt [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        fn boun<caret>dary(self: Rectangle) -> u64 {
+---
+    <sel>impl RectangleGeometry of ShapeGeometry<Rectangle> {
+        fn boundary(self: Rectangle) -> u64 {
+            2 * (self.height + self.width)
+        }
+        fn area(self: Rectangle) -> u64 {
+            self.height * self.width
+        }
+    }</sel>
```
Comment (piotmag769) [inline tests/test_data/goto/items.txt:163]: Idk if that's correct, imo it should go to the function in the trait definition

File: tests/e2e/goto_definition.rs (hunk 1 above)
Comment (mkaput) [reply tests/e2e/goto_definition.rs:95]: my next unpublished PR will add a `None` variant - for inline macro exprs (#116)

File: tests/test_data/goto/items.txt (hunk 3 above)
Comment (mkaput) [reply tests/test_data/goto/items.txt:163]: I agree with you, but this problem is out of scope of this PR and actually applies to almost all items

File: tests/test_data/goto/enum_variants.txt (hunk 2 above)
Comment (mkaput) [reply tests/test_data/goto/enum_variants.txt:32]:
```cairo
enum Foo {
    Bar, // here
    Baz,
}
```

Comment (mkaput) [reply tests/test_data/goto/enum_variants.txt:32]:
```cairo
enum Foo {
    Bar,
    Baz, // here
}
```

**Review by Arcticae — APPROVED** [review]

File: tests/test_data/goto/items.txt (hunk 3 above)
Comment (piotmag769) [reply tests/test_data/goto/items.txt:163]: Do we have an issue for that?

File: tests/e2e/goto_definition.rs (hunk 1 above)
Comment (piotmag769) [reply tests/e2e/goto_definition.rs:95]: Please refrain from doing it in the future 🙏 It makes it so:
1. Changes are not self-contained
2. Review is actually harder (I have to predict the future/look up the next PR)

File: tests/test_data/goto/enum_variants.txt (hunk 2 above)
Comment (mkaput) [reply tests/test_data/goto/enum_variants.txt]: fixed

Comment (mkaput) [reply tests/test_data/goto/enum_variants.txt]: fixed

File: tests/test_data/goto/items.txt (hunk 3 above)
Comment (mkaput) [reply tests/test_data/goto/items.txt:163]: #129

File: tests/e2e/goto_definition.rs (hunk 1 above)
Comment (mkaput) [reply tests/e2e/goto_definition.rs:95]: sorry, thought this will be clear to you

**Review by piotmag769 — APPROVED** [review]

**Pull request merged**